pkg/sink/kafka/sarama.go

```diff
@@ -167,3 +184,66 @@ func completeSaramaSASLConfig(ctx context.Context, config *sarama.Config, o *Opt
 
 	return nil
 }
+
+func GetKafkaVersion(ctx context.Context, config *sarama.Config, o *Options) (sarama.KafkaVersion, error) {
+	var err error
+	addrs := o.BrokerEndpoints
+	if len(addrs) > 1 {
+		// Shuffle the list of addresses to randomize the order in which
+		// connections are attempted. This prevents routing all connections
+		// to the first broker (which will usually succeed).
+		rand.Shuffle(len(addrs), func(i, j int) {
+			addrs[i], addrs[j] = addrs[j], addrs[i]
+		})
+	}
+	for i := range addrs {
+		broker := sarama.NewBroker(addrs[i])
+		err = broker.Open(config)
+		if err != nil {
+			log.Warn("Kafka fail to open broker", zap.String("addr", addrs[i]))
+			continue
+		}
+		defer func() {
+			broker.Close()
+		}()
+		apiResponse, err := broker.ApiVersions(&sarama.ApiVersionsRequest{})
+		if err != nil {
+			log.Warn("Kafka fail to get ApiVersions", zap.String("addr", addrs[i]))
+			continue
+		}
+		// ApiKey method
+		// 0      Produce
+		// 3      Metadata (default)
+		version := apiResponse.ApiKeys[3].MaxVersion
+		kafkaVersion := requiredVersion(version)
+		return kafkaVersion, nil
+	}
+	return sarama.V2_0_0_0, err
+}
+
+func requiredVersion(version int16) sarama.KafkaVersion {
```

Comment:
`requiredVersion` is not a good name here, this function is  just a version mapping.